crates/red_knot_python_semantic/src/types.rs

```diff
@@ -756,6 +756,8 @@ impl<'db> Type<'db> {
                         },
                     )
             }
+            (Type::Never, Type::Never) => false,
```

Comment:
I think this case is a) wrong (because `Never` should be assignable to `Never`) and b) never reached, due to the `is_equivalent_to` check that takes precedence above.

There is a test verifying that `Never` is actually assignable to `Never`.
```suggestion
```